x-pack/legacy/plugins/apm/public/hooks/useFetcher.tsx

```diff
@@ -74,14 +74,22 @@ export function useFetcher<Response>(
       dispatchStatus({ id, isLoading: false });
       didCancel = true;
     };
-  }, [...effectKey, counter]);
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [
+    counter,
+    id,
+    preservePreviousResponse,
+    dispatchStatus,
+    ...effectKey
+    /* eslint-enable react-hooks/exhaustive-deps */
```

Comment:
Is it complaining about `counter` not being used?

Reply:
I guess that's fine. Although would be better if we could avoid disabling the lint rule somehow...

Reply:
Yeah, and we can't add `fn` because it's usually recreated every render. Passing fn as a dependency would be more correct, but it would mean that we require the consumer to wrap the function that is passed to `useFetcher` in  `useCallback()`. Adding `result.data` breaks stuff as well. I figure we have to do a significant rewrite of this hook to have correct dependencies.

Reply:
Okay, let's just leave it at this, then